app/src/main/java/org/eyeseetea/malariacare/data/remote/api/UserAccountAPIDataSource.java

```diff
@@ -0,0 +1,129 @@
+package org.eyeseetea.malariacare.data.remote.api;
+
+import android.util.Log;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.eyeseetea.malariacare.data.database.model.UserDB;
+import org.eyeseetea.malariacare.data.IUserAccountDataSource;
+import org.eyeseetea.malariacare.domain.entity.Credentials;
+import org.eyeseetea.malariacare.domain.entity.UserAccount;
+import org.eyeseetea.malariacare.domain.exception.PullApiParsingException;
+import org.eyeseetea.malariacare.domain.usecase.UserFilter;
+import org.eyeseetea.malariacare.utils.DateParser;
+import org.json.JSONObject;
+
+import java.util.Date;
+
+public class UserAccountAPIDataSource extends OkHttpClientDataSource implements IUserAccountDataSource {
+
+    private static final String DHIS_PULL_API="/api/";
+
+    private static String QUERY_USER_ATTRIBUTES =
+            "/%s?fields=name,userCredentials[username],attributeValues[value,attribute[code]]id&paging=false";
+
+
+    private static final String TAG = ".PullDhisApiDataSource";
+
+    public static final String EVENT = "event";
+    public static final String ATTRIBUTEVALUES = "attributeValues";
+    public static final String ATTRIBUTE = "attribute";
+    public static final String VALUE = "value";
+    public static final String CODE = "code";
+    private static final String USER = "users";
+    private static final String NAME = "name";
+    private static final String USERCREDENTIALS = "userCredentials";
+    private static final String USERNAME = "username";
+
+    public UserAccountAPIDataSource(Credentials credentials) {
+        super(credentials);
+    }
+
+    @Override
+    public UserAccount getUser(UserFilter userFilter)  {
+        String username = "";
+        String name = "";
+        String announcement;
+        String closedDateAsString;
+        String data = USER + String.format(QUERY_USER_ATTRIBUTES, userFilter.getUid());
+        Log.d(TAG, String.format("getUserAttributesApiCall(%s) -> %s", USER, data));
+        try {
+            String response = executeCall(DHIS_PULL_API+data);
+            JsonNode jsonNode = parseResponse(response);
+            JsonNode jsonNodeArray = jsonNode.get(ATTRIBUTEVALUES);
+            String newMessage = "";
+            String closeDate = "";
+            for (int i = 0; i < jsonNodeArray.size(); i++) {
+                newMessage =
+                        getUserAnnouncement(jsonNodeArray, newMessage, i,
+                                UserDB.ATTRIBUTE_USER_ANNOUNCEMENT);
+                closeDate = getUserCloseDate(jsonNodeArray, closeDate, i);
+            }
+            announcement =newMessage;
+            closedDateAsString=closeDate;
+            username = getUsername(jsonNode);
+            name = getName(jsonNode);
+        } catch (Exception ex) {
+            Log.e(TAG, "Cannot read user last updated from server with");
```

Comment:
We should not return null UserAccount to avoid work with nulls, we should throw the exception and if in network first throw an exception then the repository should return local data